### PR TITLE
Fja 5 new fjaa delivery model

### PR DESCRIPTION
**Pull request by alireid**
Hi Chris, as discussed, new logic and various other bits to facilitate this

### CODE REVIEW

**Review by chrisfoster76 — APPROVED** [review]
Couple of unused classes but approving ready

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Cohort/ApprenticeRequestToSelectDeliveryModelViewModelMapper.cs [hunk 1]
```diff
@@ -4,19 +4,34 @@
...
+            long accountLegalEntityId = 0;
+            string encodedAccountId = string.Empty;
+
+            if (_authorizationService.IsAuthorized(EmployerFeature.FJAA))
+            {
+                accountLegalEntityId = source.AccountLegalEntityId;
+                encodedAccountId = source.AccountHashedId;
+            }
+
+            var response = await _approvalsApiClient.GetProviderCourseDeliveryModels(source.ProviderId, source.CourseCode, encodedAccountId, accountLegalEntityId);
```
[the remark this answers is not in the document]
Comment (alireid) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Mappers/Cohort/ApprenticeRequestToSelectDeliveryModelViewModelMapper.cs]: Cleaned that one away

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Apprentice/ConfirmEditApprenticeship.cshtml [hunk 2]
```diff
@@ -56,7 +56,7 @@
                     <dt class="govuk-summary-list__key"> Unique learner number </dt>
                     <dd class="govuk-summary-list__value"> @Model.OriginalApprenticeship.ULN</dd>
                 </div>
-                @if (Model.OriginalApprenticeship.DeliveryModel == DeliveryModel.PortableFlexiJob)
+                @if (Model.OriginalApprenticeship.DeliveryModel == DeliveryModel.PortableFlexiJob || Model.OriginalApprenticeship.DeliveryModel == DeliveryModel.FlexiJobAgency)
```
[the remark this answers is not in the document]
Comment (alireid) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Apprentice/ConfirmEditApprenticeship.cshtml:59]: Good shout - made that change

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Cohort/Details.cshtml [hunk 3]
```diff
@@ -73,11 +73,16 @@
 
         @foreach (var course in Model.Courses)
         {
-            <h3 class="govuk-heading-m">@course.Count x @course.DisplayCourseName 
+            <h3 class="govuk-heading-m">
+                @course.Count x @course.DisplayCourseName
                 @if (@course.DeliveryModel == DeliveryModel.PortableFlexiJob)
```
[the remark this answers is not in the document]
Comment (alireid) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Views/Cohort/Details.cshtml:78]: Good shout - made that change

File: src/SFA.DAS.EmployerCommitmentsV2/Services/Approvals/Responses/GetAgencyResponse.cs [hunk 4]
```diff
@@ -0,0 +1,7 @@
+﻿namespace SFA.DAS.EmployerCommitmentsV2.Services.Approvals.Responses
+{
+    public class GetAgencyResponse
+    {
+        public int LegalEntityId { get; set; }
```
[the remark this answers is not in the document]
Comment (alireid) [reply src/SFA.DAS.EmployerCommitmentsV2/Services/Approvals/Responses/GetAgencyResponse.cs]: Good shout - removed!

File: src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Agency/Agency.cs [hunk 5]
```diff
@@ -0,0 +1,7 @@
+﻿namespace SFA.DAS.EmployerCommitmentsV2.Web.Models.Agency
+{
+    public class Agency
+    {
+        public int LegalEntityId { get; set; }
```
[the remark this answers is not in the document]
Comment (alireid) [reply src/SFA.DAS.EmployerCommitmentsV2.Web/Models/Agency/Agency.cs:5]: Nope! Nuked!

**Pull request merged**